### PR TITLE
Adding a new template to deploy phabricator on Ubuntu

### PR DESCRIPTION
**Pull request by julienstroheker**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
Adding a new template to deploy phabricator on Ubuntu

### Description of the change
Adding a new template to deploy phabricator on Ubuntu

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@julienstroheker__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@julienstroheker__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

File: phabricator-on-ubuntu/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Install Phabricator on a Ubuntu Virtual Machine using Custom Script Linux Extension
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fjulienstroheker%2Fazure-quickstart-templates%2Fmaster%2Fphabricator-on-ubuntu%2Fazuredeploy.json" target="_blank">
```
Comment (singhkay) [inline phabricator-on-ubuntu/README.md]: Links need to point to Azure location

File: phabricator-on-ubuntu/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,226 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: auto-gen the storage account name. See #944

File: phabricator-on-ubuntu/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+      "type": "string",
+      "metadata":{
+        "description": "Storage account name"
+      }
+    },
+    "dnsNameForPublicIP": {
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: See #966 for dns label guidance

File: phabricator-on-ubuntu/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        "description": "Admin UserName for the Linux VM"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata":{
+        "description": "Admin Password for the Linux VM"
+      }
+    },
+    "location": {
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: Use resourceGroup().location instead where location is needed

File: phabricator-on-ubuntu/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_A2",
+      "metadata":{
+        "description": "Size for the VM"
+      }
+    },
+    "publicIPAddressName": {
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: This doesn't need to be a parameter and can become a variable

File: phabricator-on-ubuntu/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+      }
+    },
+    "vmName": {
+      "type": "string",
+      "defaultValue": "myLinuxVM",
+      "metadata":{
+        "description": "VM name for the resource inside Azure"
+      }
+    },
+    "virtualNetworkName": {
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: This doesn't need to be a parameter and can become a variable

File: phabricator-on-ubuntu/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+      "type": "string",
+      "defaultValue": "myVNET",
+      "metadata":{
+        "description": "VNet name for the resource inside Azure"
+      }
+    },
+    "nicName": {
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: This doesn't need to be a parameter and can become a variable

File: phabricator-on-ubuntu/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+    "imageOffer":"UbuntuServer",
+    "imageSKU":"16.04.0-LTS",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',parameters('virtualNetworkName'))]",
+    "subnet1Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: Use GA API version 2015-06-15

File: phabricator-on-ubuntu/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+      "name": "[concat(parameters('vmName'),'/installphabricator')]",
+      "apiVersion": "2015-05-01-preview",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', parameters('vmName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.2",
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: Use latest custom script and autoupgrademinorversion: "true" property

File: phabricator-on-ubuntu/phabricator-install-ubuntu.sh [hunk 10]
```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
```
Comment (singhkay) [inline phabricator-on-ubuntu/phabricator-install-ubuntu.sh]: Please organize scripts and images according to recommended approach shown here https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#files-folders-and-naming-conventions

**Comment by julienstroheker** [comment]
@singhkay Thanks you very much for the advises and the review. 
I reviewed all the code with your guidance and seems working on my side, let me know if I missed something.

File: phabricator-on-ubuntu/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,190 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "dnsNameForPublicIP": {
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: Please call this dnsLabelPrefix to set the right expectations with the users as in #966

File: phabricator-on-ubuntu/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('storageAccountName')]",
+      "apiVersion": "2015-06-15",
+      "location": "[variables('location')]",
+      "properties": {
+        "accountType": "[variables('storageAccountType')]"
+      }
+    },
+    {
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline phabricator-on-ubuntu/azuredeploy.json]: Use 2015-06-15 api version for all compute, storage and network resources

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**